src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -11,35 +11,32 @@ namespace Microsoft.ML
     /// <summary>
     /// Specifies input and output column names for a transformation.
     /// </summary>
-    [BestFriend]
-    internal sealed class ColumnOptions
+    public sealed class InputOutputColumnPair
     {
-        private readonly string _outputColumnName;
-        private readonly string _inputColumnName;
+        /// <summary>
+        /// Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref="OutputColumnName"/> will be used as source.
+        /// </summary>
+        public readonly string InputColumnName;
```

Comment:
>InputColumnName [](start = 31, length = 15)

Should these be properties?

Reply:
Ok making these properties!

---
In reply to: [268761458](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268761458) [](ancestors = 268761458)